contracts/VAIController.sol

```diff
@@ -67,17 +67,54 @@ contract VAIController is VAIControllerStorageG2, VAIControllerErrorReporter, Ex
      */
     event MintFee(address minter, uint feeAmount);
 
+    /**
+     * @notice Emitted when set vai mint old capped Amount
+     */
+    event SetMintCappedOldAmount(address admin, uint256 oldVaiMintCappedAmount);
+
+    /**
+     * @notice Emitted when set vai mint new capped Amount
+     */
+    event SetMintCappedNewAmount(address admin, uint256 newVaiMintCappedAmount);
+
     /*** Main Actions ***/
     struct MintLocalVars {
         Error err;
         MathError mathErr;
         uint mintAmount;
     }
 
+    /**
+     * @notice Set vai mint capped amount
+     */
+    function setMintCappedAmount(uint newMintCappedAmount) public returns (uint) {
+        // Check caller is admin
+        if (!(msg.sender == admin)) {
+            return fail(Error.UNAUTHORIZED, FailureInfo.SET_VAI_MINT_CAPPED_CHECK);
+        }
+        // Emit the event with old amount
+        emit SetMintCappedOldAmount(msg.sender, mintCappedAmount);
+        // Set new vai mint capped amount
+        mintCappedAmount = newMintCappedAmount;
+        // Emit the event with new amount
+        emit SetMintCappedNewAmount(msg.sender, newMintCappedAmount);
+
+        return uint(Error.NO_ERROR);
+    }
+
     function mintVAI(uint mintVAIAmount) external nonReentrant returns (uint) {
         if(address(comptroller) != address(0)) {
+            MathError mErr;
+            uint sumVaiMintAmount;
+
             require(mintVAIAmount > 0, "mintVAIAmount cannt be zero");
 
+            // Check vai mint capped
+            uint vaiTotalAmount = VAI(getVAIAddress()).totalSupply();
+            (mErr, sumVaiMintAmount) = addUInt(vaiTotalAmount, mintVAIAmount);
+            require(mErr == MathError.NO_ERROR, "SUM_VAI_MINT_AMOUNT_CALCULATION_FAILED");
+            require(mintCappedAmount >= sumVaiMintAmount, "vai mint amount over capped");
```

Comment:
@fengliu93 if cap amount is zero, lets not check the mint cap. https://github.com/VenusProtocol/venus-protocol/blob/48c7400f668116f5e5fa92299e28cf16b49b485c/contracts/Comptroller.sol#L387